tests/h/traversal/annotation_test.py

```diff
@@ -9,340 +9,108 @@
-
-    def test_it_assigns_create_permission_to_authenticated_request(
-        self, set_permissions, pyramid_request
-    ):
-        set_permissions(
-            "acct:adminuser@foo", principals=[pyramid.security.Authenticated]
-        )
-
-        context = AnnotationRoot(pyramid_request)
-
-        assert pyramid_request.has_permission(Permission.Annotation.CREATE, context)
-
-    def test_get_item_fetches_annotation(self, pyramid_request, storage):
-        factory = AnnotationRoot(pyramid_request)
-
-        factory["123"]
-        storage.fetch_annotation.assert_called_once_with(pyramid_request.db, "123")
-
-    def test_get_item_returns_annotation_context(self, pyramid_request, storage):
-        factory = AnnotationRoot(pyramid_request)
-        storage.fetch_annotation.return_value = mock.Mock()
-
-        resource = factory["123"]
-        assert isinstance(resource, AnnotationContext)
-
-    def test_get_item_resource_has_right_annotation(self, pyramid_request, storage):
-        factory = AnnotationRoot(pyramid_request)
-        storage.fetch_annotation.return_value = mock.Mock()
+        root = AnnotationRoot(pyramid_request)
 
-        resource = factory["123"]
-        assert resource.annotation == storage.fetch_annotation.return_value
+        assert permits(root, [security.Authenticated], Permission.Annotation.CREATE)
+        assert not permits(root, [], Permission.Annotation.CREATE)
 
-    def test_get_item_raises_when_annotation_is_not_found(
-        self, storage, pyramid_request
-    ):
-        factory = AnnotationRoot(pyramid_request)
-        storage.fetch_annotation.return_value = None
-
-        with pytest.raises(KeyError):
-            factory["123"]
-
-    def test_get_item_has_right_group_service(
-        self, pyramid_request, storage, groupfinder_service
+    def test_annotation_lookup(
+        self,
+        pyramid_request,
+        AnnotationContext,
+        storage,
+        groupfinder_service,
+        links_service,
     ):
-        factory = AnnotationRoot(pyramid_request)
-        storage.fetch_annotation.return_value = mock.Mock()
+        root = AnnotationRoot(pyramid_request)
 
-        resource = factory["123"]
-        assert resource.group_service == groupfinder_service
+        context = root[sentinel.annotation_id]
 
-    def test_get_item_has_right_links_service(
-        self, pyramid_request, storage, links_service
-    ):
-        factory = AnnotationRoot(pyramid_request)
-        storage.fetch_annotation.return_value = mock.Mock()
-
-        resource = factory["123"]
-        assert resource.links_service == links_service
+        assert context == AnnotationContext.return_value
+        storage.fetch_annotation.assert_called_once_with(
+            pyramid_request.db, sentinel.annotation_id
+        )
+        AnnotationContext.assert_called_once_with(
+            storage.fetch_annotation.return_value, groupfinder_service, links_service
+        )
```

Comment:
This is another case of putting every assertion in a different test. These are very tedious plumbing type tests and we should get them over with as quickly as we can.

Apart from being very long, it's kind of impossible to follow these. When put into one test we can see a chain of:

 * This was called with this
 * The next object was fed the previous one's return value
 * ...
 * Finally we get the return value of this